utils/linux/proc_events.py

```diff
@@ -0,0 +1,214 @@
+"""Callbacks for process events
+
+This module uses the Linux's Process Connector mechanism to invoke registered callbacks on various process events.
+To achieve this, once the first callback is registered a thread is invoked to listen to these events and execute the
+registered callbacks.
+
+This interface with the kernel isn't too well documented, but to learn more about it you can read this blog post:
+https://nick-black.com/dankwiki/index.php/The_Proc_Connector_and_Socket_Filters
+
+
+TODO: Add more callbacks.
+TODO: Use socket filter.
```

Comment:
Wdym here btw?

Reply:
Filter the messages we're getting so the thread won't get up for every event in the system

Reply:
Ah, like, a BPF filter for specific PIDs?

Reply:
Yes.
I'm somewhat concerned that this thread will introduce noticeable overhead. This filter will solve it.

Reply:
https://www.explainxkcd.com/wiki/index.php/1691:_Optimization